lib/autoload/globals/Iter.ngs

```diff
@@ -373,3 +373,48 @@ F skip(i:Iter, count:Int){
 TEST [3, 4, 5, 34, 5].Iter().skip(3).Arr() == [34, 5] 
 TEST [3, 40, 15, 34, 5].Iter().skip(2).Arr() == [15, 34, 5] 
 TEST {[3, 40, 15, 34, 5].Iter().skip(10)}.assert(SkipError)
+
+doc Type that allows us to iterate over a directory.
+type DirIter(Iter)
+
+# ---------- DirIter ----------
+doc Calls DirIter constructor.
+doc %RET - DirIter which allows us to iterate over a directory
+F Iter(d:Dir) DirIter(d)
+
+doc Initialize the DirIter with specific Dir
+F init(i:DirIter, d:Dir){
+	i.val = d
+	i.paths = dir(d)
```

Comment:
Don't have much time now but this is surely a big issue. Don't load everything at once into memory. The iterator should be smarter than that. It should use `c_readdir()` as necessary. Think about `Dir(...).Iter().limit(5)`.

Reply:
I do not know if I understood the point completly. 
But, taking look at F dir(...) in stdlibs.ngs, I could see that it is using both c_readdir() and c_opendir(). Would not be the same using c_readdir directly for loading in this method?

Reply:
`c_readdir()` must be used in all cases. The difference is when and how many times it is called. If you need to return an `Arr` you must read the whole directory in any case. If you are returning an `Iter`, `c_readdir()` should be called when necessary, that is when `next()` is called. There is no point reading the whole directory. Imagine `next()` will be called just one time or not at all.